{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "384ae40c",
   "metadata": {},
   "source": [
    "# Filtering Nanopore data and assigning taxonomy to reads using Epi2Me\n",
    "\n",
    "There are a variety of tools that can be used to filter out bad quality Nanopore reads and reads with unsuitable lengths, and to match reads to their most likely species. However, we're going to focus on a very easy set of tools to use: [Epi2me labs](https://labs.epi2me.io). This is a Nanopore bioinformatics workbench that Oxford Nanopore runs, with a variety of workflows that utilize already-existing tools.\n",
    "\n",
    "In our case, we're going to be using the metagenomics workflow. This workflow allows you filter by read length and quality, filter out sequences from a host genome, detect antimicrobial resistance genes (in the case of shotgun metagenomic sequencing) and assign taxonomy to reads. There are two options for how to assign taxonomy - one is using Kraken2/Bracken, which uses kmers to map reads to the lowest common ancestor (LCA) that matches that pattern of kmers and predicts which descendant is the best match (Kraken2) and converts this into species abundances (Bracken). Below is a visualization of the Kraken workflow.\n",
    "\n",
    "The other option is mapping the reads to a provided database of specific genomes using minimap. This is useful if you know what should be in your sample already - but that's rarely the case for microbiome research."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fb4db027",
   "metadata": {},
   "source": [
    "### Kraken workflow\n",
    "\n",
    "<img src=\"https://media.springernature.com/full/springer-static/image/art%3A10.1186%2Fgb-2014-15-3-r46/MediaObjects/13059_2013_Article_3351_Fig1_HTML.jpg?as=webp\" alt=\"Alternative text\"  width=\"800\" height=\"500\" />\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ed39ecb7",
   "metadata": {},
   "source": [
    "### Using Epi2me Labs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0be93dd3",
   "metadata": {},
   "source": [
    "Epi2me Labs has a [desktop application](https://labs.epi2me.io/downloads/) that is very easy and intuitive to use. However, it's not possible to use the desktop app with the way these computing resources are set up, so we're going to use it in its command-line application (it's kind of easier to see all your options this way anyway)\n",
    "\n",
    "Epi2me workflows run using nextflow, which is a python-based method of chaining multiple programs together. It works with docker, a method of creating computing environments as packages. It's not important to know these things, but it IS important to know that I've pre-installed them here, and you will need to install them yourself if you want to run Epi2me Labs on your computer.\n",
    "\n",
    "Run the command below to see all the options for the metagenomics workflow. In this case we need to use the flag \"--help\" rather than \"-h\" as \"-h\" will bring up a more general menu."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "cdbe8163",
   "metadata": {
    "scrolled": false,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N E X T F L O W  ~  version 23.04.3\n",
      "NOTE: Your local project version looks outdated - a different revision is available in the remote repository [4397263427]\n",
      "Launching `https://github.com/epi2me-labs/wf-metagenomics` [reverent_euclid] DSL2 - revision: 54e6d3e743 [master]\n",
      "\u001b[33mWARN: NEXTFLOW RECURSION IS A PREVIEW FEATURE - SYNTAX AND FUNCTIONALITY CAN CHANGE IN FUTURE RELEASES\u001b[39m\u001b[K\n",
      "\n",
      "\u001b[0;92m||||||||||   \u001b[0m\u001b[2m_____ ____ ___ ____  __  __ _____      _       _\n",
      "\u001b[0;92m||||||||||  \u001b[0m\u001b[2m| ____|  _ \\_ _|___ \\|  \\/  | ____|    | | __ _| |__  ___\n",
      "\u001b[0;33m|||||       \u001b[0m\u001b[2m|  _| | |_) | |  __) | |\\/| |  _| _____| |/ _` | '_ \\/ __|\n",
      "\u001b[0;33m|||||       \u001b[0m\u001b[2m| |___|  __/| | / __/| |  | | |__|_____| | (_| | |_) \\__ \\\n",
      "\u001b[0;94m||||||||||  \u001b[0m\u001b[2m|_____|_|  |___|_____|_|  |_|_____|    |_|\\__,_|_.__/|___/\n",
      "\u001b[0;94m||||||||||  \u001b[0m\u001b[1mwf-metagenomics v2.5.0-g54e6d3e\u001b[0m\n",
      "\u001b[2m--------------------------------------------------------------------------------\u001b[0m\n",
      "Typical pipeline command:\n",
      "\n",
      "  \u001b[0;36mnextflow run epi2me-labs/wf-metagenomics \\ \n",
      "\t--fastq 'wf-metagenomics-demo/test_data' \\ \n",
      "\t--threads 2\u001b[0m\n",
      "\n",
      "\u001b[4m\u001b[1mInput Options\u001b[0m\n",
      "  --fastq                       \u001b[2m[string]  \u001b[0mA fastq file or a directory of directories containing fastq input files.\u001b[2m\u001b[0m\n",
      "  --classifier                  \u001b[2m[choice]  \u001b[0mKraken2 or Minimap2 workflow to be used for classification of reads.\u001b[2m [default: kraken2]\u001b[0m\n",
      "                                          * kraken2\n",
      "                                          * minimap2\n",
      "  --batch_size                  \u001b[2m[integer] \u001b[0mMaximum number of sequence records to process in a batch.\u001b[2m\u001b[0m\n",
      "  --analyse_unclassified        \u001b[2m[boolean] \u001b[0mAnalyse unclassified reads from input directory. By default the workflow will not process reads in the unclassified \n",
      "                                          directory.\u001b[2m\u001b[0m \n",
      "  --exclude_host                \u001b[2m[string]  \u001b[0mA FASTA or MMI file of the host reference. Reads that align with this reference will be excluded from the \n",
      "                                          analysis.\u001b[2m\u001b[0m \n",
      "\n",
      "\u001b[4m\u001b[1mReal Time Analysis Options\u001b[0m\n",
      "  --watch_path                  \u001b[2m[boolean] \u001b[0mEnable to continuously watch the input directory for new input files. Reads will be analysed as they appear\u001b[2m\u001b[0m\n",
      "  --read_limit                  \u001b[2m[integer] \u001b[0mStop processing data when a particular number of reads have been analysed. By default the workflow will run \n",
      "                                          indefinitely.\u001b[2m\u001b[0m \n",
      "\n",
      "\u001b[4m\u001b[1mSample Options\u001b[0m\n",
      "  --sample_sheet                \u001b[2m[string]  \u001b[0mA CSV file used to map barcodes to sample aliases. The sample sheet can be provided when the input data is a directory \n",
      "                                          containing sub-directories with FASTQ files. Used only with the minimap2 classifier.\u001b[2m\u001b[0m \n",
      "  --sample                      \u001b[2m[string]  \u001b[0mA single sample name for non-multiplexed data. Permissible if passing a single .fastq(.gz) file or directory of \n",
      "                                          .fastq(.gz) files.\u001b[2m\u001b[0m \n",
      "\n",
      "\u001b[4m\u001b[1mReference Options\u001b[0m\n",
      "  --database_set                \u001b[2m[choice]  \u001b[0mSets the reference, databases and taxonomy datasets that will be used for classifying reads. Choices: \n",
      "                                          ['ncbi_16s_18s','ncbi_16s_18s_28s_ITS', 'SILVA_138_1', 'PlusPF-8', 'PlusPFP-8']. Workflow will require memory \n",
      "                                          available to be slightly higher than the size of the database. PlusPF-8 database requires more than 8Gb.\u001b[2m [default: \n",
      "                                          ncbi_16s_18s]\u001b[0m \n",
      "                                          * ncbi_16s_18s\n",
      "                                          * ncbi_16s_18s_28s_ITS\n",
      "                                          * SILVA_138_1\n",
      "                                          * PlusPF-8\n",
      "                                          * PlusPFP-8\n",
      "  --database                    \u001b[2m[string]  \u001b[0mNot required but can be used to specifically override kraken2 database [.tar.gz or Directory].\u001b[2m\u001b[0m\n",
      "  --taxonomy                    \u001b[2m[string]  \u001b[0mNot required but can be used to specifically override taxonomy database. Change the default to use a different \n",
      "                                          taxonomy file  [.tar.gz or directory].\u001b[2m\u001b[0m \n",
      "  --store_dir                   \u001b[2m[string]  \u001b[0mWhere to store initial download of database.\u001b[2m [default: store_dir]\u001b[0m\n",
      "  --reference                   \u001b[2m[string]  \u001b[0mOverride the FASTA reference file selected by the database_set parameter. It can be a FASTA format reference sequence \n",
      "                                          collection or a minimap2 MMI format index.\u001b[2m\u001b[0m \n",
      "  --ref2taxid                   \u001b[2m[string]  \u001b[0mNot required but can be used to specify a  ref2taxid mapping. Format is .tsv (refname  taxid), no header row.\u001b[2m\u001b[0m\n",
      "  --taxonomic_rank              \u001b[2m[choice]  \u001b[0mReturns results at the taxonomic rank chosen. In the kraken2 pipeline: set the level that bracken will estimate \n",
      "                                          abundance at. Default: S (species). Other possible options are K (kingdom level), P (phylum), C (class), O (order), F \n",
      "                                          (family), and G (genus).\u001b[2m [default: S]\u001b[0m \n",
      "                                          * S\n",
      "                                          * G\n",
      "                                          * k\n",
      "                                          * F\n",
      "                                          * O\n",
      "                                          * C\n",
      "                                          * P\n",
      "\n",
      "\u001b[4m\u001b[1mKraken2 Options\u001b[0m\n",
      "  --bracken_dist                \u001b[2m[string]  \u001b[0mSpecifically override bracken kmer dist file\u001b[2m\u001b[0m\n",
      "  --bracken_length              \u001b[2m[integer] \u001b[0mSet the length value bracken will use\u001b[2m\u001b[0m\n",
      "  --port                        \u001b[2m[integer] \u001b[0mNetwork port for communication between kraken2 server and clients\u001b[2m [default: 8080]\u001b[0m\n",
      "  --host                        \u001b[2m[string]  \u001b[0mNetwork hostname (or IP address) for communication between kraken2 server and clients. (See also 'external_kraken2' \n",
      "                                          parameter).\u001b[2m [default: localhost]\u001b[0m \n",
      "  --external_kraken2            \u001b[2m[boolean] \u001b[0mWhether a pre-existing kraken2 server should be used, rather than creating one as part of the workflow\u001b[2m\u001b[0m\n",
      "  --kraken2_memory_mapping      \u001b[2m[boolean] \u001b[0mAvoids loading database into RAM\u001b[2m\u001b[0m\n",
      "  --include_kraken2_assignments \u001b[2m[boolean] \u001b[0mA per sample TSV file that indicates how each input sequence was classified as well as the taxon that has been \n",
      "                                          assigned to each read. The TSV's will only be output on completion of the workflow and therefore not at all if using \n",
      "                                          the real time option whilst running indefinitely.\u001b[2m\u001b[0m \n",
      "  --kraken2_confidence          \u001b[2m[number]  \u001b[0mKraken2 Confidence score threshold. Default: 0.0. Valid interval: 0-1\u001b[2m\u001b[0m\n",
      "\n",
      "\u001b[4m\u001b[1mMinimap2 Options\u001b[0m\n",
      "  --minimap2filter              \u001b[2m[string]  \u001b[0mFilter output of minimap2 by taxids inc. child nodes, E.g. \"9606,1404\"\u001b[2m\u001b[0m\n",
      "  --minimap2exclude             \u001b[2m[boolean] \u001b[0mInvert minimap2filter and exclude the given taxids instead\u001b[2m\u001b[0m\n",
      "  --split_prefix                \u001b[2m[boolean] \u001b[0mEnable if using a very large reference with minimap2\u001b[2m\u001b[0m\n",
      "  --keep_bam                    \u001b[2m[boolean] \u001b[0mCopy bam files into the output directory.\u001b[2m\u001b[0m\n",
      "\n",
      "\u001b[4m\u001b[1mAntimicrobial Resistance Options\u001b[0m\n",
      "  --amr                         \u001b[2m[boolean] \u001b[0mScan reads for antimicrobial resistance or virulence genes\u001b[2m\u001b[0m\n",
      "  --amr_db                      \u001b[2m[choice]  \u001b[0mDatabase of antimicrobial resistance or virulence genes to use.\u001b[2m [default: resfinder]\u001b[0m\n",
      "                                          * resfinder\n",
      "                                          * ecoli_vf\n",
      "                                          * plasmidfinder\n",
      "                                          * card\n",
      "                                          * argannot\n",
      "                                          * vfdb\n",
      "                                          * ncbi\n",
      "                                          * megares\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                          * ecoh\n",
      "  --amr_minid                   \u001b[2m[integer] \u001b[0mThreshold of required identity to report a match between a gene in the database and fastq reads. Valid interval: \n",
      "                                          0-100\u001b[2m [default: 80]\u001b[0m \n",
      "  --amr_mincov                  \u001b[2m[integer] \u001b[0mMinimum coverage (breadth-of) threshold required to report a match between a gene in the  database and fastq reads. \n",
      "                                          Valid interval: 0-100\u001b[2m [default: 80]\u001b[0m \n",
      "\n",
      "\u001b[4m\u001b[1mReport Options\u001b[0m\n",
      "  --abundance_threshold         \u001b[2m[number]  \u001b[0mRemove those taxa whose abundance is lower than the chosen value.\u001b[2m\u001b[0m\n",
      "  --n_taxa_barplot              \u001b[2m[integer] \u001b[0mNumber of most abundance taxa to be displayed in the barplot. The rest of taxa will be grouped under the \"Other\" \n",
      "                                          category.\u001b[2m [default: 8]\u001b[0m \n",
      "\n",
      "\u001b[4m\u001b[1mOutput Options\u001b[0m\n",
      "  --out_dir                     \u001b[2m[string]  \u001b[0mDirectory for output of all user-facing files.\u001b[2m [default: output]\u001b[0m\n",
      "\n",
      "\u001b[4m\u001b[1mAdvanced Options\u001b[0m\n",
      "  --min_len                     \u001b[2m[integer] \u001b[0mSpecify read length lower limit.\u001b[2m\u001b[0m\n",
      "  --min_read_qual               \u001b[2m[number]  \u001b[0mSpecify read quality lower limit.\u001b[2m\u001b[0m\n",
      "  --max_len                     \u001b[2m[integer] \u001b[0mSpecify read length upper limit\u001b[2m\u001b[0m\n",
      "  --threads                     \u001b[2m[integer] \u001b[0mMaximum number of CPU threads to use per workflow task.\u001b[2m [default: 2]\u001b[0m\n",
      "  --server_threads              \u001b[2m[integer] \u001b[0mNumber of CPU threads used by the kraken2 server for classifying reads.\u001b[2m [default: 2]\u001b[0m\n",
      "  --kraken_clients              \u001b[2m[integer] \u001b[0mNumber of clients that can connect at once to the kraken-server for classifying reads.\u001b[2m [default: 2]\u001b[0m\n",
      "\n",
      "\u001b[4m\u001b[1mOther parameters\u001b[0m\n",
      "  --monochrome_logs             \u001b[2m[boolean] \u001b[0mnull\u001b[2m\u001b[0m\n",
      "  --validate_params             \u001b[2m[boolean] \u001b[0mnull\u001b[2m [default: true]\u001b[0m\n",
      "  --show_hidden_params          \u001b[2m[boolean] \u001b[0mnull\u001b[2m\u001b[0m\n",
      "\n",
      "\u001b[2m!! Hiding 6 params, use --show_hidden_params to show them !!\n",
      "\u001b[0m\u001b[2m--------------------------------------------------------------------------------\u001b[0m\n",
      "If you use epi2me-labs/wf-metagenomics for your analysis please cite:\n",
      "\n",
      "* The nf-core framework\n",
      "  https://doi.org/10.1038/s41587-020-0439-x\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "nextflow run epi2me-labs/wf-metagenomics --help"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4beb82af",
   "metadata": {},
   "source": [
    "Let's try running one of the mock datasets."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "a8e88a46-9c3d-4dfb-8f55-a66c0ad90e5e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ls ~/data/\n",
      "Jessica_Zymo_DNA_16S_flongle_test  Zymo_16S_SRR25400687_Nanoplot_2\n",
      "SRR25400687_unfiltered.biom\t   Zymo_16S_table.biom\n",
      "Zymo_16S_SRR25400687.fastq\t   zymo_standard.csv\n",
      "Zymo_16S_SRR25400687_Nanoplot\t   zymo_standard_bacteriaonly.csv\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls ~/data/"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5f6c92f4",
   "metadata": {},
   "source": [
    "Let's run the Institut Pasteur de Lille 16S dataset through Epi2me's amplicon taxonomy workflow. We'll need to provide the program with the input data (The fastq file that we input into Nanoplot), the name of a directory to store our results, and the name of the database that Epi2me (and therefore the program Kraken2 mentioned above) should use for assigning taxonomy. Since we are using 16S sequences (recall from the lectures yesterday that this is the bacterial 'barcode' gene), we should use a database of 16S sequences. We'll use the database 'ncbi_16s_18s'."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "573215f5",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N E X T F L O W  ~  version 23.04.3\n",
      "NOTE: Your local project version looks outdated - a different revision is available in the remote repository [4397263427]\n",
      "Launching `https://github.com/epi2me-labs/wf-metagenomics` [modest_joliot] DSL2 - revision: 54e6d3e743 [master]\n",
      "\u001b[33mWARN: NEXTFLOW RECURSION IS A PREVIEW FEATURE - SYNTAX AND FUNCTIONALITY CAN CHANGE IN FUTURE RELEASES\u001b[39m\u001b[K\n",
      "\n",
      "\u001b[0;92m||||||||||   \u001b[0m\u001b[2m_____ ____ ___ ____  __  __ _____      _       _\n",
      "\u001b[0;92m||||||||||  \u001b[0m\u001b[2m| ____|  _ \\_ _|___ \\|  \\/  | ____|    | | __ _| |__  ___\n",
      "\u001b[0;33m|||||       \u001b[0m\u001b[2m|  _| | |_) | |  __) | |\\/| |  _| _____| |/ _` | '_ \\/ __|\n",
      "\u001b[0;33m|||||       \u001b[0m\u001b[2m| |___|  __/| | / __/| |  | | |__|_____| | (_| | |_) \\__ \\\n",
      "\u001b[0;94m||||||||||  \u001b[0m\u001b[2m|_____|_|  |___|_____|_|  |_|_____|    |_|\\__,_|_.__/|___/\n",
      "\u001b[0;94m||||||||||  \u001b[0m\u001b[1mwf-metagenomics v2.5.0-g54e6d3e\u001b[0m\n",
      "\u001b[2m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1mCore Nextflow options\u001b[0m\n",
      "  \u001b[0;34mrevision       : \u001b[0;32mmaster\u001b[0m\n",
      "  \u001b[0;34mrunName        : \u001b[0;32mmodest_joliot\u001b[0m\n",
      "  \u001b[0;34mcontainerEngine: \u001b[0;32mdocker\u001b[0m\n",
      "  \u001b[0;34mlaunchDir      : \u001b[0;32m/Users/asimpson/Documents/GitHub/WSSU-Nanopore-Workshop-Fall2023\u001b[0m\n",
      "  \u001b[0;34mworkDir        : \u001b[0;32m/Users/asimpson/Documents/GitHub/WSSU-Nanopore-Workshop-Fall2023/work\u001b[0m\n",
      "  \u001b[0;34mprojectDir     : \u001b[0;32m/Users/asimpson/.nextflow/assets/epi2me-labs/wf-metagenomics\u001b[0m\n",
      "  \u001b[0;34muserName       : \u001b[0;32masimpson\u001b[0m\n",
      "  \u001b[0;34mprofile        : \u001b[0;32mstandard\u001b[0m\n",
      "  \u001b[0;34mconfigFiles    : \u001b[0;32m/Users/asimpson/.nextflow/assets/epi2me-labs/wf-metagenomics/nextflow.config\u001b[0m\n",
      "\n",
      "\u001b[1mInput Options\u001b[0m\n",
      "  \u001b[0;34mfastq          : \u001b[0;32mdata/Zymo_16S_SRR25400687.fastq\u001b[0m\n",
      "\n",
      "\u001b[1mReference Options\u001b[0m\n",
      "  \u001b[0;34mdatabase_sets  : \u001b[0;32m[ncbi_16s_18s:[reference:https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_16s_18s.fna, refindex:https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_16s_18s.fna.fai, database:https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_kraken2.tar.gz, kmer_dist:https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/database1000mers.kmer_distrib, ref2taxid:https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ref2taxid.targloci.tsv, taxonomy:https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump_archive/taxdmp_2023-01-01.zip], ncbi_16s_18s_28s_ITS:[reference:https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s_28s_ITS/ncbi_16s_18s_28s_ITS.fna, refindex:https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s_28s_ITS/ncbi_16s_18s_28s_ITS.fna.fai, database:https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s_28s_ITS/ncbi_16s_18s_28s_ITS_kraken2.tar.gz, kmer_dist:https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s_28s_ITS/database1000mers.kmer_distrib, ref2taxid:https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s_28s_ITS/ref2taxid.ncbi_16s_18s_28s_ITS.tsv, taxonomy:https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump_archive/taxdmp_2023-01-01.zip], SILVA_138_1:[database:null], PlusPF-8:[database:https://genome-idx.s3.amazonaws.com/kraken/k2_pluspf_08gb_20230314.tar.gz, taxonomy:https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump_archive/new_taxdump_2023-03-01.zip], PlusPFP-8:[database:https://genome-idx.s3.amazonaws.com/kraken/k2_pluspfp_08gb_20230314.tar.gz, taxonomy:https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump_archive/new_taxdump_2023-03-01.zip]]\u001b[0m\n",
      "\n",
      "\u001b[1mOutput Options\u001b[0m\n",
      "  \u001b[0;34mout_dir        : \u001b[0;32mEpi2me_results/Zymo_16S_SRR25400687_Epi2me\u001b[0m\n",
      "\n",
      "!! Only displaying parameters that differ from the pipeline defaults !!\n",
      "\u001b[2m--------------------------------------------------------------------------------\u001b[0m\n",
      "If you use epi2me-labs/wf-metagenomics for your analysis please cite:\n",
      "\n",
      "* The nf-core framework\n",
      "  https://doi.org/10.1038/s41587-020-0439-x\n",
      "\n",
      "\n",
      "\u001b[2m--------------------------------------------------------------------------------\u001b[0m\n",
      "This is epi2me-labs/wf-metagenomics v2.5.0-g54e6d3e.\n",
      "\u001b[2m--------------------------------------------------------------------------------\u001b[0m\n",
      "Checking inputs.\n",
      "Checking fastq input.\n",
      "[-        ] process > fastcat                        -\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackTaxonomy -\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackDatabase -\u001b[K\n",
      "[-        ] process > kraken_pipeline:determine_b... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken_server  -\u001b[K\n",
      "[-        ] process > kraken_pipeline:run_common:... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:run_common:... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[-        ] process > kraken_pipeline:output         -\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "\u001b[15A\n",
      "executor >  local (3)\u001b[K\n",
      "[b7/3aed0e] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackTaxonomy -\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackDatabase -\u001b[K\n",
      "[-        ] process > kraken_pipeline:determine_b... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken_server  -\u001b[K\n",
      "[38/b6e6d8] process > kraken_pipeline:run_common:... [  0%] 0 of 1\u001b[K\n",
      "[cf/6ae75c] process > kraken_pipeline:run_common:... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[-        ] process > kraken_pipeline:output         -\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "\u001b[16A\n",
      "executor >  local (3)\u001b[K\n",
      "[b7/3aed0e] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackTaxonomy -\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackDatabase -\u001b[K\n",
      "[-        ] process > kraken_pipeline:determine_b... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken_server  -\u001b[K\n",
      "[38/b6e6d8] process > kraken_pipeline:run_common:... [  0%] 0 of 1\u001b[K\n",
      "[cf/6ae75c] process > kraken_pipeline:run_common:... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[-        ] process > kraken_pipeline:output         -\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "Staging foreign file: https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump_archive/taxdmp_2023-01-01.zip\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_kraken2.tar.gz\u001b[K\n",
      "\u001b[18A\n",
      "executor >  local (3)\u001b[K\n",
      "[b7/3aed0e] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackTaxonomy -\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackDatabase -\u001b[K\n",
      "[-        ] process > kraken_pipeline:determine_b... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken_server  -\u001b[K\n",
      "[38/b6e6d8] process > kraken_pipeline:run_common:... [  0%] 0 of 1\u001b[K\n",
      "[cf/6ae75c] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[-        ] process > kraken_pipeline:output         [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "Staging foreign file: https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump_archive/taxdmp_2023-01-01.zip\u001b[K\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_kraken2.tar.gz\u001b[K\n",
      "\u001b[18A\n",
      "executor >  local (4)\u001b[K\n",
      "[b7/3aed0e] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackTaxonomy -\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackDatabase -\u001b[K\n",
      "[-        ] process > kraken_pipeline:determine_b... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken_server  -\u001b[K\n",
      "[38/b6e6d8] process > kraken_pipeline:run_common:... [  0%] 0 of 1\u001b[K\n",
      "[cf/6ae75c] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[d4/fe696f] process > kraken_pipeline:output (1)     [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "Staging foreign file: https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump_archive/taxdmp_2023-01-01.zip\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_kraken2.tar.gz\u001b[K\n",
      "\u001b[18A\n",
      "executor >  local (4)\u001b[K\n",
      "[b7/3aed0e] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackTaxonomy -\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackDatabase -\u001b[K\n",
      "[-        ] process > kraken_pipeline:determine_b... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken_server  -\u001b[K\n",
      "[38/b6e6d8] process > kraken_pipeline:run_common:... [  0%] 0 of 1\u001b[K\n",
      "[cf/6ae75c] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[d4/fe696f] process > kraken_pipeline:output (1)     [100%] 1 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "Staging foreign file: https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump_archive/taxdmp_2023-01-01.zip\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_kraken2.tar.gz\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/database1000mers.kmer_distrib\u001b[K\n",
      "\u001b[19A\n",
      "executor >  local (5)\u001b[K\n",
      "[b7/3aed0e] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackTaxonomy -\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackDatabase -\u001b[K\n",
      "[-        ] process > kraken_pipeline:determine_b... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken_server  -\u001b[K\n",
      "[38/b6e6d8] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[cf/6ae75c] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[ca/98d128] process > kraken_pipeline:output (2)     [ 50%] 1 of 2\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "Staging foreign file: https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump_archive/taxdmp_2023-01-01.zip\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_kraken2.tar.gz\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/database1000mers.kmer_distrib\u001b[K\n",
      "\u001b[19A\n",
      "executor >  local (5)\u001b[K\n",
      "[b7/3aed0e] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackTaxonomy -\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackDatabase -\u001b[K\n",
      "[-        ] process > kraken_pipeline:determine_b... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken_server  -\u001b[K\n",
      "[38/b6e6d8] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[cf/6ae75c] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[ca/98d128] process > kraken_pipeline:output (2)     [100%] 2 of 2\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "Staging foreign file: https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump_archive/taxdmp_2023-01-01.zip\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_kraken2.tar.gz\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/database1000mers.kmer_distrib\u001b[K\n",
      "\u001b[19A\n",
      "executor >  local (6)\u001b[K\n",
      "[b7/3aed0e] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackTaxonomy -\u001b[K\n",
      "[b1/d5b4a4] process > kraken_pipeline:unpackDatabase [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:determine_b... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken_server  -\u001b[K\n",
      "[38/b6e6d8] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[cf/6ae75c] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[ca/98d128] process > kraken_pipeline:output (2)     [100%] 2 of 2\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "Staging foreign file: https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump_archive/taxdmp_2023-01-01.zip\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_kraken2.tar.gz\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/database1000mers.kmer_distrib\u001b[K\n",
      "\u001b[19A\n",
      "executor >  local (8)\u001b[K\n",
      "[b7/3aed0e] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackTaxonomy -\u001b[K\n",
      "[b1/d5b4a4] process > kraken_pipeline:unpackDatabase [100%] 1 of 1 ✔\u001b[K\n",
      "[db/502c5c] process > kraken_pipeline:determine_b... [  0%] 0 of 1\u001b[K\n",
      "[e0/5937cf] process > kraken_pipeline:kraken_server  [  0%] 0 of 1\u001b[K\n",
      "[38/b6e6d8] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[cf/6ae75c] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[ca/98d128] process > kraken_pipeline:output (2)     [100%] 2 of 2\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "Staging foreign file: https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump_archive/taxdmp_2023-01-01.zip\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_kraken2.tar.gz\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/database1000mers.kmer_distrib\u001b[K\n",
      "\u001b[19A\n",
      "executor >  local (8)\u001b[K\n",
      "[b7/3aed0e] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackTaxonomy -\u001b[K\n",
      "[b1/d5b4a4] process > kraken_pipeline:unpackDatabase [100%] 1 of 1 ✔\u001b[K\n",
      "[db/502c5c] process > kraken_pipeline:determine_b... [  0%] 0 of 1\u001b[K\n",
      "[e0/5937cf] process > kraken_pipeline:kraken_server  [  0%] 0 of 1\u001b[K\n",
      "[38/b6e6d8] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[cf/6ae75c] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[ca/98d128] process > kraken_pipeline:output (2)     [100%] 2 of 2\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "Staging foreign file: https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump_archive/taxdmp_2023-01-01.zip\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_kraken2.tar.gz\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/database1000mers.kmer_distrib\u001b[K\n",
      "\u001b[19A\n",
      "executor >  local (8)\u001b[K\n",
      "[b7/3aed0e] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackTaxonomy -\u001b[K\n",
      "[b1/d5b4a4] process > kraken_pipeline:unpackDatabase [100%] 1 of 1 ✔\u001b[K\n",
      "[db/502c5c] process > kraken_pipeline:determine_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[e0/5937cf] process > kraken_pipeline:kraken_server  [  0%] 0 of 1\u001b[K\n",
      "[38/b6e6d8] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[cf/6ae75c] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[ca/98d128] process > kraken_pipeline:output (2)     [100%] 2 of 2\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "Staging foreign file: https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump_archive/taxdmp_2023-01-01.zip\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_kraken2.tar.gz\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/database1000mers.kmer_distrib\u001b[K\n",
      "\u001b[19A\n",
      "executor >  local (9)\u001b[K\n",
      "[b7/3aed0e] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[cf/cdbaa6] process > kraken_pipeline:unpackTaxonomy [  0%] 0 of 1\u001b[K\n",
      "[b1/d5b4a4] process > kraken_pipeline:unpackDatabase [100%] 1 of 1 ✔\u001b[K\n",
      "[db/502c5c] process > kraken_pipeline:determine_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[e0/5937cf] process > kraken_pipeline:kraken_server  [  0%] 0 of 1\u001b[K\n",
      "[38/b6e6d8] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[cf/6ae75c] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[ca/98d128] process > kraken_pipeline:output (2)     [100%] 2 of 2\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[19A\n",
      "executor >  local (9)\u001b[K\n",
      "[b7/3aed0e] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[cf/cdbaa6] process > kraken_pipeline:unpackTaxonomy [100%] 1 of 1 ✔\u001b[K\n",
      "[b1/d5b4a4] process > kraken_pipeline:unpackDatabase [100%] 1 of 1 ✔\u001b[K\n",
      "[db/502c5c] process > kraken_pipeline:determine_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[e0/5937cf] process > kraken_pipeline:kraken_server  [  0%] 0 of 1\u001b[K\n",
      "[38/b6e6d8] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[cf/6ae75c] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[ca/98d128] process > kraken_pipeline:output (2)     [100%] 2 of 2\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[19A\n",
      "executor >  local (10)\u001b[K\n",
      "[b7/3aed0e] process > fastcat (1)                    [100%] 1 of 1 ✔\u001b[K\n",
      "[cf/cdbaa6] process > kraken_pipeline:unpackTaxonomy [100%] 1 of 1 ✔\u001b[K\n",
      "[b1/d5b4a4] process > kraken_pipeline:unpackDatabase [100%] 1 of 1 ✔\u001b[K\n",
      "[db/502c5c] process > kraken_pipeline:determine_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[e0/5937cf] process > kraken_pipeline:kraken_server  [  0%] 0 of 1\u001b[K\n",
      "[38/b6e6d8] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[cf/6ae75c] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[3c/0138db] process > kraken_pipeline:kraken2_cli... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[ca/98d128] process > kraken_pipeline:output (2)     [100%] 2 of 2\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "\n",
      "\u001b[16A\n",
      "executor >  local (10)\u001b[K\n",
      "[b7/3aed0e] process > fastcat (1)                    [100%] 1 of 1 ✔\u001b[K\n",
      "[cf/cdbaa6] process > kraken_pipeline:unpackTaxonomy [100%] 1 of 1 ✔\u001b[K\n",
      "[b1/d5b4a4] process > kraken_pipeline:unpackDatabase [100%] 1 of 1 ✔\u001b[K\n",
      "[db/502c5c] process > kraken_pipeline:determine_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[e0/5937cf] process > kraken_pipeline:kraken_server  [  0%] 0 of 1\u001b[K\n",
      "[38/b6e6d8] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[cf/6ae75c] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[3c/0138db] process > kraken_pipeline:kraken2_cli... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[ca/98d128] process > kraken_pipeline:output (2)     [100%] 2 of 2\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "WARN: Killing running tasks (2)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "nextflow run epi2me-labs/wf-metagenomics --fastq data/Zymo_16S_SRR25400687.fastq \\\n",
    "    --out_dir Epi2me_results/Zymo_16S_Epi2me --database_set 'ncbi_16s_18s'\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "12f0c64f-7f52-428b-9774-59a4502c8411",
   "metadata": {},
   "source": [
    "Let's navigate to the results file in the file menu on the left and see what our results look like."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cf62d40a-aa03-4154-95d9-74bcdce2c2f9",
   "metadata": {},
   "source": [
    "### Quality filtering and length filtering "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "85759762",
   "metadata": {},
   "source": [
    "Epi2me's workflow provides some options for filtering the data by length and by the quality score of the reads. What difference does filtering by length, by quality, or both, make in terms of what species we detect?"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "06cad1ae-1af0-4159-b81f-a63e06ef75b4",
   "metadata": {
    "tags": []
   },
   "source": [
    "  --min_len                     [integer] Specify read length lower limit.\n",
    "  \n",
    "  --min_read_qual               [number]  Specify read quality lower limit.\n",
    "  \n",
    "  --max_len                     [integer] Specify read length upper limit"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "42ba9fc2",
   "metadata": {},
   "source": [
    "#### Filtering by length \n",
    "\n",
    "We know from yesterday's discussiong that the 16S gene should be around 1,500bp. Let's try filtering the data so that the only sequences we keep are between 1,400 and 1,700 bp."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5952ee65-953b-473c-8129-4969fc3d55fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "nextflow run epi2me-labs/wf-metagenomics --fastq data/Zymo_16S_SRR25400687.fastq \\\n",
    "    --out_dir Epi2me_results/Zymo_16S_Epi2me_lengthfiltered --database_set 'ncbi_16s_18s'\\\n",
    "    --min_len 1400 --max_len 1700\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4c422bfe",
   "metadata": {},
   "source": [
    "#### Filtering by quality\n",
    "\n",
    "Now let's try filtering out all the sequences with an average quality score below 14. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "394a0e97",
   "metadata": {},
   "outputs": [],
   "source": [
    "nextflow run epi2me-labs/wf-metagenomics --fastq data/Zymo_16S_SRR25400687.fastq \\\n",
    "    --out_dir Epi2me_results/Zymo_16S_Epi2me_qualityfiltered --database_set 'ncbi_16s_18s'\\\n",
    "    --min_read_qual 14\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d4ce3bff",
   "metadata": {},
   "source": [
    "#### Filtering by both"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "28046397-6011-43ca-adda-f6cedd849d3f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "nextflow run epi2me-labs/wf-metagenomics --fastq ~/data/Zymo_16S_SRR25400687.fastq \\\n",
      "    --out_dir ~/backup/Zymo_16S_SRR25400687_Epi2me_filtered --database_set 'ncbi_16s_18s'\\\n",
      "    --min_len 1390 --max_len 1780 --min_read_qual 14\n",
      "N E X T F L O W  ~  version 23.04.3\n",
      "Launching `https://github.com/epi2me-labs/wf-metagenomics` [astonishing_hodgkin] DSL2 - revision: 54e6d3e743 [master]\n",
      "\u001b[33mWARN: NEXTFLOW RECURSION IS A PREVIEW FEATURE - SYNTAX AND FUNCTIONALITY CAN CHANGE IN FUTURE RELEASES\u001b[39m\u001b[K\n",
      "\n",
      "\u001b[0;92m||||||||||   \u001b[0m\u001b[2m_____ ____ ___ ____  __  __ _____      _       _\n",
      "\u001b[0;92m||||||||||  \u001b[0m\u001b[2m| ____|  _ \\_ _|___ \\|  \\/  | ____|    | | __ _| |__  ___\n",
      "\u001b[0;33m|||||       \u001b[0m\u001b[2m|  _| | |_) | |  __) | |\\/| |  _| _____| |/ _` | '_ \\/ __|\n",
      "\u001b[0;33m|||||       \u001b[0m\u001b[2m| |___|  __/| | / __/| |  | | |__|_____| | (_| | |_) \\__ \\\n",
      "\u001b[0;94m||||||||||  \u001b[0m\u001b[2m|_____|_|  |___|_____|_|  |_|_____|    |_|\\__,_|_.__/|___/\n",
      "\u001b[0;94m||||||||||  \u001b[0m\u001b[1mwf-metagenomics v2.5.0-g54e6d3e\u001b[0m\n",
      "\u001b[2m--------------------------------------------------------------------------------\u001b[0m\n",
      "\u001b[1mCore Nextflow options\u001b[0m\n",
      "  \u001b[0;34mrevision       : \u001b[0;32mmaster\u001b[0m\n",
      "  \u001b[0;34mrunName        : \u001b[0;32mastonishing_hodgkin\u001b[0m\n",
      "  \u001b[0;34mcontainerEngine: \u001b[0;32mdocker\u001b[0m\n",
      "  \u001b[0;34mlaunchDir      : \u001b[0;32m/home/gl4u/Notebooks\u001b[0m\n",
      "  \u001b[0;34mworkDir        : \u001b[0;32m/home/gl4u/Notebooks/work\u001b[0m\n",
      "  \u001b[0;34mprojectDir     : \u001b[0;32m/home/gl4u/.nextflow/assets/epi2me-labs/wf-metagenomics\u001b[0m\n",
      "  \u001b[0;34muserName       : \u001b[0;32mgl4u\u001b[0m\n",
      "  \u001b[0;34mprofile        : \u001b[0;32mstandard\u001b[0m\n",
      "  \u001b[0;34mconfigFiles    : \u001b[0;32m/home/gl4u/.nextflow/assets/epi2me-labs/wf-metagenomics/nextflow.config\u001b[0m\n",
      "\n",
      "\u001b[1mInput Options\u001b[0m\n",
      "  \u001b[0;34mfastq          : \u001b[0;32m/home/gl4u/data/Zymo_16S_SRR25400687.fastq\u001b[0m\n",
      "\n",
      "\u001b[1mReference Options\u001b[0m\n",
      "  \u001b[0;34mdatabase_sets  : \u001b[0;32m[ncbi_16s_18s:[reference:https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_16s_18s.fna, refindex:https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_16s_18s.fna.fai, database:https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_kraken2.tar.gz, kmer_dist:https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/database1000mers.kmer_distrib, ref2taxid:https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ref2taxid.targloci.tsv, taxonomy:https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump_archive/taxdmp_2023-01-01.zip], ncbi_16s_18s_28s_ITS:[reference:https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s_28s_ITS/ncbi_16s_18s_28s_ITS.fna, refindex:https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s_28s_ITS/ncbi_16s_18s_28s_ITS.fna.fai, database:https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s_28s_ITS/ncbi_16s_18s_28s_ITS_kraken2.tar.gz, kmer_dist:https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s_28s_ITS/database1000mers.kmer_distrib, ref2taxid:https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s_28s_ITS/ref2taxid.ncbi_16s_18s_28s_ITS.tsv, taxonomy:https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump_archive/taxdmp_2023-01-01.zip], SILVA_138_1:[database:null], PlusPF-8:[database:https://genome-idx.s3.amazonaws.com/kraken/k2_pluspf_08gb_20230314.tar.gz, taxonomy:https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump_archive/new_taxdump_2023-03-01.zip], PlusPFP-8:[database:https://genome-idx.s3.amazonaws.com/kraken/k2_pluspfp_08gb_20230314.tar.gz, taxonomy:https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump_archive/new_taxdump_2023-03-01.zip]]\u001b[0m\n",
      "\n",
      "\u001b[1mOutput Options\u001b[0m\n",
      "  \u001b[0;34mout_dir        : \u001b[0;32m/home/gl4u/backup/Zymo_16S_SRR25400687_Epi2me_filtered\u001b[0m\n",
      "\n",
      "\u001b[1mAdvanced Options\u001b[0m\n",
      "  \u001b[0;34mmin_len        : \u001b[0;32m1390\u001b[0m\n",
      "  \u001b[0;34mmin_read_qual  : \u001b[0;32m14\u001b[0m\n",
      "  \u001b[0;34mmax_len        : \u001b[0;32m1780\u001b[0m\n",
      "\n",
      "!! Only displaying parameters that differ from the pipeline defaults !!\n",
      "\u001b[2m--------------------------------------------------------------------------------\u001b[0m\n",
      "If you use epi2me-labs/wf-metagenomics for your analysis please cite:\n",
      "\n",
      "* The nf-core framework\n",
      "  https://doi.org/10.1038/s41587-020-0439-x\n",
      "\n",
      "\n",
      "\u001b[2m--------------------------------------------------------------------------------\u001b[0m\n",
      "This is epi2me-labs/wf-metagenomics v2.5.0-g54e6d3e.\n",
      "\u001b[2m--------------------------------------------------------------------------------\u001b[0m\n",
      "Checking inputs.\n",
      "Checking fastq input.\n",
      "[-        ] process > fastcat -\u001b[K\n",
      "\u001b[2A\n",
      "executor >  local (1)\u001b[K\n",
      "[-        ] process > fastcat                        [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackTaxonomy -\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackDatabase -\u001b[K\n",
      "[-        ] process > kraken_pipeline:determine_b... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken_server  -\u001b[K\n",
      "[de/5e770b] process > kraken_pipeline:run_common:... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:run_common:... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[-        ] process > kraken_pipeline:output         -\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "\u001b[16A\n",
      "executor >  local (3)\u001b[K\n",
      "[cb/761cce] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackTaxonomy -\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackDatabase -\u001b[K\n",
      "[-        ] process > kraken_pipeline:determine_b... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken_server  -\u001b[K\n",
      "[de/5e770b] process > kraken_pipeline:run_common:... [  0%] 0 of 1\u001b[K\n",
      "[42/840b41] process > kraken_pipeline:run_common:... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[-        ] process > kraken_pipeline:output         -\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "\u001b[16A\n",
      "executor >  local (4)\u001b[K\n",
      "[cb/761cce] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackTaxonomy -\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackDatabase -\u001b[K\n",
      "[-        ] process > kraken_pipeline:determine_b... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken_server  -\u001b[K\n",
      "[de/5e770b] process > kraken_pipeline:run_common:... [  0%] 0 of 1\u001b[K\n",
      "[42/840b41] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[90/28329c] process > kraken_pipeline:output (1)     [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "\u001b[16A\n",
      "executor >  local (4)\u001b[K\n",
      "[cb/761cce] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackTaxonomy -\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackDatabase -\u001b[K\n",
      "[-        ] process > kraken_pipeline:determine_b... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken_server  -\u001b[K\n",
      "[de/5e770b] process > kraken_pipeline:run_common:... [  0%] 0 of 1\u001b[K\n",
      "[42/840b41] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[90/28329c] process > kraken_pipeline:output (1)     [100%] 1 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "\u001b[16A\n",
      "executor >  local (4)\u001b[K\n",
      "[cb/761cce] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackTaxonomy [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackDatabase -\u001b[K\n",
      "[-        ] process > kraken_pipeline:determine_b... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken_server  -\u001b[K\n",
      "[de/5e770b] process > kraken_pipeline:run_common:... [  0%] 0 of 1\u001b[K\n",
      "[42/840b41] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[90/28329c] process > kraken_pipeline:output (1)     [100%] 1 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "[skipping] Stored process > kraken_pipeline:unpackTaxonomy\u001b[K\n",
      "\u001b[17A\n",
      "executor >  local (5)\u001b[K\n",
      "[cb/761cce] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackTaxonomy [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackDatabase -\u001b[K\n",
      "[-        ] process > kraken_pipeline:determine_b... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken_server  -\u001b[K\n",
      "[de/5e770b] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[42/840b41] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[39/9237bb] process > kraken_pipeline:output (2)     [ 50%] 1 of 2\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "[skipping] Stored process > kraken_pipeline:unpackTaxonomy\u001b[K\n",
      "\u001b[17A\n",
      "executor >  local (5)\u001b[K\n",
      "[cb/761cce] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackTaxonomy [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackDatabase -\u001b[K\n",
      "[-        ] process > kraken_pipeline:determine_b... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken_server  -\u001b[K\n",
      "[de/5e770b] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[42/840b41] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[39/9237bb] process > kraken_pipeline:output (2)     [ 50%] 1 of 2\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "[skipping] Stored process > kraken_pipeline:unpackTaxonomy\u001b[K\n",
      "\u001b[17A\n",
      "executor >  local (5)\u001b[K\n",
      "[cb/761cce] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackTaxonomy [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackDatabase -\u001b[K\n",
      "[-        ] process > kraken_pipeline:determine_b... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken_server  -\u001b[K\n",
      "[de/5e770b] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[42/840b41] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[39/9237bb] process > kraken_pipeline:output (2)     [100%] 2 of 2\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "[skipping] Stored process > kraken_pipeline:unpackTaxonomy\u001b[K\n",
      "\u001b[17A\n",
      "executor >  local (5)\u001b[K\n",
      "[cb/761cce] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackTaxonomy [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:unpackDatabase -\u001b[K\n",
      "[-        ] process > kraken_pipeline:determine_b... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken_server  -\u001b[K\n",
      "[de/5e770b] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[42/840b41] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[39/9237bb] process > kraken_pipeline:output (2)     [100%] 2 of 2\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "[skipping] Stored process > kraken_pipeline:unpackTaxonomy\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_kraken2.tar.gz\u001b[K\n",
      "\u001b[18A\n",
      "executor >  local (7)\u001b[K\n",
      "[cb/761cce] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackTaxonomy [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackDatabase [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[3c/fd30ed] process > kraken_pipeline:determine_b... [  0%] 0 of 1\u001b[K\n",
      "[39/13390a] process > kraken_pipeline:kraken_server  [  0%] 0 of 1\u001b[K\n",
      "[de/5e770b] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[42/840b41] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[39/9237bb] process > kraken_pipeline:output (2)     [100%] 2 of 2\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "[skipping] Stored process > kraken_pipeline:unpackTaxonomy\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_kraken2.tar.gz\u001b[K\n",
      "[skipping] Stored process > kraken_pipeline:unpackDatabase\u001b[K\n",
      "\u001b[19A\n",
      "executor >  local (7)\u001b[K\n",
      "[cb/761cce] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackTaxonomy [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackDatabase [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[3c/fd30ed] process > kraken_pipeline:determine_b... [  0%] 0 of 1\u001b[K\n",
      "[39/13390a] process > kraken_pipeline:kraken_server  [  0%] 0 of 1\u001b[K\n",
      "[de/5e770b] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[42/840b41] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[39/9237bb] process > kraken_pipeline:output (2)     [100%] 2 of 2\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "[skipping] Stored process > kraken_pipeline:unpackTaxonomy\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_kraken2.tar.gz\u001b[K\n",
      "[skipping] Stored process > kraken_pipeline:unpackDatabase\u001b[K\n",
      "\u001b[19A\n",
      "executor >  local (7)\u001b[K\n",
      "[cb/761cce] process > fastcat (1)                    [  0%] 0 of 1\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackTaxonomy [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackDatabase [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[3c/fd30ed] process > kraken_pipeline:determine_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[39/13390a] process > kraken_pipeline:kraken_server  [  0%] 0 of 1\u001b[K\n",
      "[de/5e770b] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[42/840b41] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:kraken2_client -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[39/9237bb] process > kraken_pipeline:output (2)     [100%] 2 of 2\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "[skipping] Stored process > kraken_pipeline:unpackTaxonomy\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_kraken2.tar.gz\u001b[K\n",
      "[skipping] Stored process > kraken_pipeline:unpackDatabase\u001b[K\n",
      "\u001b[19A\n",
      "executor >  local (8)\u001b[K\n",
      "[cb/761cce] process > fastcat (1)                    [100%] 1 of 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackTaxonomy [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackDatabase [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[3c/fd30ed] process > kraken_pipeline:determine_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[39/13390a] process > kraken_pipeline:kraken_server  [  0%] 0 of 1\u001b[K\n",
      "[de/5e770b] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[42/840b41] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[1b/9bf636] process > kraken_pipeline:kraken2_cli... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[39/9237bb] process > kraken_pipeline:output (2)     [100%] 2 of 2\u001b[K\n",
      "[-        ] process > kraken_pipeline:stop_kraken... -\u001b[K\n",
      "[skipping] Stored process > kraken_pipeline:unpackTaxonomy\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_kraken2.tar.gz\u001b[K\n",
      "[skipping] Stored process > kraken_pipeline:unpackDatabase\u001b[K\n",
      "\u001b[19A\n",
      "executor >  local (11)\u001b[K\n",
      "[cb/761cce] process > fastcat (1)                    [100%] 1 of 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackTaxonomy [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackDatabase [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[3c/fd30ed] process > kraken_pipeline:determine_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[39/13390a] process > kraken_pipeline:kraken_server  [  0%] 0 of 1\u001b[K\n",
      "[de/5e770b] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[42/840b41] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[1b/9bf636] process > kraken_pipeline:kraken2_cli... [100%] 1 of 1 ✔\u001b[K\n",
      "[34/dab171] process > kraken_pipeline:progressive... [  0%] 0 of 1\u001b[K\n",
      "[29/913b52] process > kraken_pipeline:progressive... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[39/9237bb] process > kraken_pipeline:output (2)     [100%] 2 of 2\u001b[K\n",
      "[ea/327b5d] process > kraken_pipeline:stop_kraken... [  0%] 0 of 1\u001b[K\n",
      "[skipping] Stored process > kraken_pipeline:unpackTaxonomy\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_kraken2.tar.gz\u001b[K\n",
      "[skipping] Stored process > kraken_pipeline:unpackDatabase\u001b[K\n",
      "\u001b[19A\n",
      "executor >  local (11)\u001b[K\n",
      "[cb/761cce] process > fastcat (1)                    [100%] 1 of 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackTaxonomy [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackDatabase [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[3c/fd30ed] process > kraken_pipeline:determine_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[39/13390a] process > kraken_pipeline:kraken_server  [  0%] 0 of 1\u001b[K\n",
      "[de/5e770b] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[42/840b41] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[1b/9bf636] process > kraken_pipeline:kraken2_cli... [100%] 1 of 1 ✔\u001b[K\n",
      "[34/dab171] process > kraken_pipeline:progressive... [  0%] 0 of 1\u001b[K\n",
      "[29/913b52] process > kraken_pipeline:progressive... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[39/9237bb] process > kraken_pipeline:output (2)     [100%] 2 of 2\u001b[K\n",
      "[ea/327b5d] process > kraken_pipeline:stop_kraken... [  0%] 0 of 1\u001b[K\n",
      "[skipping] Stored process > kraken_pipeline:unpackTaxonomy\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_kraken2.tar.gz\u001b[K\n",
      "[skipping] Stored process > kraken_pipeline:unpackDatabase\u001b[K\n",
      "\u001b[19A\n",
      "executor >  local (11)\u001b[K\n",
      "[cb/761cce] process > fastcat (1)                    [100%] 1 of 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackTaxonomy [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackDatabase [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[3c/fd30ed] process > kraken_pipeline:determine_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[39/13390a] process > kraken_pipeline:kraken_server  [100%] 1 of 1 ✔\u001b[K\n",
      "[de/5e770b] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[42/840b41] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[1b/9bf636] process > kraken_pipeline:kraken2_cli... [100%] 1 of 1 ✔\u001b[K\n",
      "[34/dab171] process > kraken_pipeline:progressive... [  0%] 0 of 1\u001b[K\n",
      "[29/913b52] process > kraken_pipeline:progressive... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:progressive... -\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[39/9237bb] process > kraken_pipeline:output (2)     [100%] 2 of 2\u001b[K\n",
      "[ea/327b5d] process > kraken_pipeline:stop_kraken... [100%] 1 of 1 ✔\u001b[K\n",
      "[skipping] Stored process > kraken_pipeline:unpackTaxonomy\u001b[K\n",
      "Staging foreign file: https://ont-exd-int-s3-euwst1-epi2me-labs.s3.amazonaws.com/wf-metagenomics/ncbi_16s_18s/ncbi_targeted_loci_kraken2.tar.gz\u001b[K\n",
      "[skipping] Stored process > kraken_pipeline:unpackDatabase\u001b[K\n",
      "\u001b[19A\n",
      "executor >  local (12)\u001b[K\n",
      "[cb/761cce] process > fastcat (1)                    [100%] 1 of 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackTaxonomy [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackDatabase [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[3c/fd30ed] process > kraken_pipeline:determine_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[39/13390a] process > kraken_pipeline:kraken_server  [100%] 1 of 1 ✔\u001b[K\n",
      "[de/5e770b] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[42/840b41] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[1b/9bf636] process > kraken_pipeline:kraken2_cli... [100%] 1 of 1 ✔\u001b[K\n",
      "[34/dab171] process > kraken_pipeline:progressive... [100%] 1 of 1 ✔\u001b[K\n",
      "[29/913b52] process > kraken_pipeline:progressive... [100%] 1 of 1 ✔\u001b[K\n",
      "[d6/2fd230] process > kraken_pipeline:progressive... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[39/9237bb] process > kraken_pipeline:output (2)     [100%] 2 of 2\u001b[K\n",
      "[ea/327b5d] process > kraken_pipeline:stop_kraken... [100%] 1 of 1 ✔\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[19A\n",
      "executor >  local (12)\u001b[K\n",
      "[cb/761cce] process > fastcat (1)                    [100%] 1 of 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackTaxonomy [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackDatabase [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[3c/fd30ed] process > kraken_pipeline:determine_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[39/13390a] process > kraken_pipeline:kraken_server  [100%] 1 of 1 ✔\u001b[K\n",
      "[de/5e770b] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[42/840b41] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[1b/9bf636] process > kraken_pipeline:kraken2_cli... [100%] 1 of 1 ✔\u001b[K\n",
      "[34/dab171] process > kraken_pipeline:progressive... [100%] 1 of 1 ✔\u001b[K\n",
      "[29/913b52] process > kraken_pipeline:progressive... [100%] 1 of 1 ✔\u001b[K\n",
      "[d6/2fd230] process > kraken_pipeline:progressive... [  0%] 0 of 1 ✔\u001b[K\n",
      "[-        ] process > kraken_pipeline:makeReport     -\u001b[K\n",
      "[39/9237bb] process > kraken_pipeline:output (2)     [100%] 2 of 2\u001b[K\n",
      "[ea/327b5d] process > kraken_pipeline:stop_kraken... [100%] 1 of 1 ✔\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[19A\n",
      "executor >  local (13)\u001b[K\n",
      "[cb/761cce] process > fastcat (1)                    [100%] 1 of 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackTaxonomy [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackDatabase [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[3c/fd30ed] process > kraken_pipeline:determine_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[39/13390a] process > kraken_pipeline:kraken_server  [100%] 1 of 1 ✔\u001b[K\n",
      "[de/5e770b] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[42/840b41] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[1b/9bf636] process > kraken_pipeline:kraken2_cli... [100%] 1 of 1 ✔\u001b[K\n",
      "[34/dab171] process > kraken_pipeline:progressive... [100%] 1 of 1 ✔\u001b[K\n",
      "[29/913b52] process > kraken_pipeline:progressive... [100%] 1 of 1 ✔\u001b[K\n",
      "[d6/2fd230] process > kraken_pipeline:progressive... [100%] 1 of 1 ✔\u001b[K\n",
      "[6f/a4fb78] process > kraken_pipeline:makeReport (1) [  0%] 0 of 1\u001b[K\n",
      "[39/9237bb] process > kraken_pipeline:output (2)     [100%] 2 of 2\u001b[K\n",
      "[ea/327b5d] process > kraken_pipeline:stop_kraken... [100%] 1 of 1 ✔\u001b[K\n",
      "\u001b[16A\n",
      "executor >  local (14)\u001b[K\n",
      "[cb/761cce] process > fastcat (1)                    [100%] 1 of 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackTaxonomy [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackDatabase [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[3c/fd30ed] process > kraken_pipeline:determine_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[39/13390a] process > kraken_pipeline:kraken_server  [100%] 1 of 1 ✔\u001b[K\n",
      "[de/5e770b] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[42/840b41] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[1b/9bf636] process > kraken_pipeline:kraken2_cli... [100%] 1 of 1 ✔\u001b[K\n",
      "[34/dab171] process > kraken_pipeline:progressive... [100%] 1 of 1 ✔\u001b[K\n",
      "[29/913b52] process > kraken_pipeline:progressive... [100%] 1 of 1 ✔\u001b[K\n",
      "[d6/2fd230] process > kraken_pipeline:progressive... [100%] 1 of 1 ✔\u001b[K\n",
      "[6f/a4fb78] process > kraken_pipeline:makeReport (1) [100%] 1 of 1 ✔\u001b[K\n",
      "[d7/426c5a] process > kraken_pipeline:output (3)     [ 66%] 2 of 3\u001b[K\n",
      "[ea/327b5d] process > kraken_pipeline:stop_kraken... [100%] 1 of 1 ✔\u001b[K\n",
      "\u001b[16A\n",
      "executor >  local (14)\u001b[K\n",
      "[cb/761cce] process > fastcat (1)                    [100%] 1 of 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackTaxonomy [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackDatabase [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[3c/fd30ed] process > kraken_pipeline:determine_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[39/13390a] process > kraken_pipeline:kraken_server  [100%] 1 of 1 ✔\u001b[K\n",
      "[de/5e770b] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[42/840b41] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[1b/9bf636] process > kraken_pipeline:kraken2_cli... [100%] 1 of 1 ✔\u001b[K\n",
      "[34/dab171] process > kraken_pipeline:progressive... [100%] 1 of 1 ✔\u001b[K\n",
      "[29/913b52] process > kraken_pipeline:progressive... [100%] 1 of 1 ✔\u001b[K\n",
      "[d6/2fd230] process > kraken_pipeline:progressive... [100%] 1 of 1 ✔\u001b[K\n",
      "[6f/a4fb78] process > kraken_pipeline:makeReport (1) [100%] 1 of 1 ✔\u001b[K\n",
      "[d7/426c5a] process > kraken_pipeline:output (3)     [100%] 3 of 3 ✔\u001b[K\n",
      "[ea/327b5d] process > kraken_pipeline:stop_kraken... [100%] 1 of 1 ✔\u001b[K\n",
      "\u001b[16A\n",
      "executor >  local (14)\u001b[K\n",
      "[cb/761cce] process > fastcat (1)                    [100%] 1 of 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackTaxonomy [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[skipped  ] process > kraken_pipeline:unpackDatabase [100%] 1 of 1, stored: 1 ✔\u001b[K\n",
      "[3c/fd30ed] process > kraken_pipeline:determine_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[39/13390a] process > kraken_pipeline:kraken_server  [100%] 1 of 1 ✔\u001b[K\n",
      "[de/5e770b] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[42/840b41] process > kraken_pipeline:run_common:... [100%] 1 of 1 ✔\u001b[K\n",
      "[1b/9bf636] process > kraken_pipeline:kraken2_cli... [100%] 1 of 1 ✔\u001b[K\n",
      "[34/dab171] process > kraken_pipeline:progressive... [100%] 1 of 1 ✔\u001b[K\n",
      "[29/913b52] process > kraken_pipeline:progressive... [100%] 1 of 1 ✔\u001b[K\n",
      "[d6/2fd230] process > kraken_pipeline:progressive... [100%] 1 of 1 ✔\u001b[K\n",
      "[6f/a4fb78] process > kraken_pipeline:makeReport (1) [100%] 1 of 1 ✔\u001b[K\n",
      "[d7/426c5a] process > kraken_pipeline:output (3)     [100%] 3 of 3 ✔\u001b[K\n",
      "[ea/327b5d] process > kraken_pipeline:stop_kraken... [100%] 1 of 1 ✔\u001b[K\n",
      "\u001b[32;1mCompleted at: 10-Sep-2023 11:11:27\n",
      "Duration    : 1m 32s\n",
      "CPU hours   : 0.1\n",
      "Succeeded   : 14\n",
      "\u001b[22;39m\u001b[K\n",
      "\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "nextflow run epi2me-labs/wf-metagenomics --fastq data/Zymo_16S_SRR25400687.fastq \\\n",
    "    --out_dir Epi2me_results/Zymo_16S_Epi2me_qualityandlengthfiltered --database_set 'ncbi_16s_18s'\\\n",
    "    --min_len 1400 --max_len 1700--min_read_qual 14\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a61e60b6",
   "metadata": {},
   "source": [
    "Now try your own way of filtering the data! Call the result file \"Epi2me_results/Zymo_16S_Epi2me_myfilter\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3fff0b20",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "9efbc003",
   "metadata": {},
   "source": [
    "Let's navigate to the left-hand menu and look at the resulting file reports together."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
