{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "11755dbf-45f0-4ca7-891e-3dd86658ce18",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Passing data to phyloseq"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "092fad9a",
   "metadata": {},
   "source": [
    "Now that we've done some taxonomic classification of our reads using different ways of filtering our sequencing data, let's get the data into a format that is more easily explored. We'll convert our data into biom files."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e892ed4a",
   "metadata": {},
   "source": [
    "What is a biom file?\n",
    "http://biom-format.org/ Known as a biological observation matrix, it's a very handy format for storing information about microbial taxa detected in metagenomic data and how many reads are assigned to each taxon. It was designed so that you can use it with a very wide variety of different applications and pipelines. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "32c1bae9",
   "metadata": {},
   "outputs": [],
   "source": [
    "We can convert the reports generated by our work"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6ec2b48b",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "kraken-biom --help"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9a286857",
   "metadata": {},
   "source": [
    "First, let's get the reports we want to convert to biom folders. These are located in the 'bracken' folder of each result file and end in \".kreport_bracken_species.txt\". Kraken2 classified our sequences, and then Bracken took the classifications and made the best guess about what species the sequence belonged to (if Kraken2 couldn't decide)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "edbcf920",
   "metadata": {},
   "source": [
    "Let's look in the contents of our Epi2me_results folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2d358d62",
   "metadata": {},
   "outputs": [],
   "source": [
    "ls Epi2me_resuls"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "523c98b1",
   "metadata": {},
   "source": [
    "There should be five folders for the five different jobs we submitted:\n",
    "\n",
    "- Zymo_16S_Epi2me\n",
    "- Zymo_16S_Epi2me_lengthfiltered\n",
    "- Zymo_16S_Epi2me_qualityfiltered\n",
    "- Zymo_16S_Epi2me_qualityandlengthfiltered\n",
    "- Zymo_16S_Epi2me_myfilter"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "83b11f69",
   "metadata": {},
   "source": [
    "Let's look in the results of the first job we ran:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "624387bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "ls Epi2me_results/Zymo_16S_Epi2me/"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "92d47fd7",
   "metadata": {},
   "source": [
    "Inside is the bracken folder, what's in there?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "447ddc53",
   "metadata": {},
   "outputs": [],
   "source": [
    "ls Epi2me_results/Zymo_16S_Epi2me/bracken"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "de81dfb5",
   "metadata": {},
   "source": [
    "Let's rename these reports and copy them to a new folder called bracken_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "222e46b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir bracken_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c462ede2",
   "metadata": {},
   "outputs": [],
   "source": [
    "echo \"meow\" > bracken_results/cow.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "47934840",
   "metadata": {},
   "outputs": [],
   "source": [
    "cp Zymo_16S_Epi2me/braken/Zymo_16S_SRR25400687.kreport_bracken_species.txt braken_results/no_filter.txt\n",
    "cp Zymo_16S_Epi2me_lengthfiltered/braken/Zymo_16S_SRR25400687.kreport_bracken_species.txt braken_results/length_filter.txt\n",
    "cp Zymo_16S_Epi2me_qualityfiltered/braken/Zymo_16S_SRR25400687.kreport_bracken_species.txt braken_results/quality_filter.txt\n",
    "cp Zymo_16S_Epi2me_qualityandlengthfiltered/braken/Zymo_16S_SRR25400687.kreport_bracken_species.txt braken_results/quality_length_filter.txt\n",
    "cp Zymo_16S_Epi2me_myfilter/braken/Zymo_16S_SRR25400687.kreport_bracken_species.txt braken_results/my_filter.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6c5c40b6-a007-4993-8d3a-e373db3fd9a5",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "ls bracken_results"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "89f95804",
   "metadata": {},
   "source": [
    "Now, let's use a program called kraken-biom (https://github.com/smdabdoub/kraken-biom) to convert our reports into one big biom file. We can put all our results into the file and each file will be treated as a separate sample."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dc326ac2",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd bracken_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3c340440",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "kraken-biom no_filter.txt length_filter.txt quality_filter.txt quality_length_filter.txt myfilter.txt \\\n",
    "    --fmt json -o ../Zymo_16S_table.biom"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
